Library/Homebrew/cmd/update.rb

```diff
@@ -277,8 +277,7 @@ def pull!(options = {})
     reset_on_interrupt { safe_system "git", *args }
 
     @current_revision = read_current_revision
-
-    if @initial_branch != @upstream_branch && !@initial_branch.empty?
+    if ARGV.homebrew_developer? && @initial_branch != @upstream_branch && !@initial_branch.empty?
```

Comment:
I think `pop_stash` is still very useful. Could we have a logical similar like this?
```
if pop stash will create conflict
  show pop stash message
else
  pop stash automatically for user
end
```